dexec/cmd.go

```diff
@@ -165,14 +170,16 @@ func (c *Cmd) Start() error {
 		if !c.DisableLogging {
 			ctx := dlog.WithField(c.ctx, "dexec.pid", c.Process.Pid)
 			dlog.Printf(ctx, "started command %q", c.Args)
-			if stdin, isFile := c.Stdin.(*os.File); isFile {
-				dlog.Printf(dlog.WithField(ctx, "dexec.stream", "stdin"), "not logging input read from file %q", stdin.Name())
-			}
-			if stdout, isFile := c.Stdout.(*os.File); isFile {
-				dlog.Printf(dlog.WithField(ctx, "dexec.stream", "stdout"), "not logging output written to file %q", stdout.Name())
-			}
-			if stderr, isFile := c.Stderr.(*os.File); isFile {
-				dlog.Printf(dlog.WithField(ctx, "dexec.stream", "stderr"), "not logging output written to file %q", stderr.Name())
+			if !c.DisableIOLogging {
+				if stdin, isFile := c.Stdin.(*os.File); isFile {
```

Comment:
It is a bit strange that what get's logged is controlled by the type of the `io.Reader`/`io.Writer` that is used. What if I decide to buffer the IO (often a good idea when working with files)? Why would that suddenly enable data logging? Or the opposite; what if I really want stdin to be logged but also want to use a `*os.File`?

This wouldn't be necessary with detailed control over what gets logged.

Reply:
> what if I really want stdin to be logged but also want to use a `*os.File`?

It's impossible.  An important use-case is being able to pass through raw file descriptors to be inherited by the child process; this is done with `*os.File`.  If we're passing through the raw file descriptors, there's no way to log the IO on that FD, we're cut out of the loop. (I *suppose* we could do something with strace/dtrace/eBPF? But, uhh...)